projects/addon-doc/components/navigation/navigation.component.ts

```diff
@@ -82,22 +80,34 @@ export class TuiDocNavigationComponent {
         readonly items: readonly TuiDocPages[],
         @Inject(TUI_DOC_SEARCH_TEXT) readonly searchText: string,
         @Inject(Router) private readonly router: Router,
-        @Inject(ActivatedRoute) private readonly activatedRoute: ActivatedRoute,
+        @Inject(ActivatedRoute) readonly activatedRoute: ActivatedRoute,
         @Self() @Inject(TuiDestroyService) private readonly destroy$: Observable<void>,
         @Inject(TUI_DOC_PAGE_LOADED)
         private readonly readyToScroll$: Observable<boolean>,
-        @Inject(TUI_DOC_SCROLL_BEHAVIOR) private readonly scrollBehavior: ScrollBehavior,
         @Inject(TUI_DOC_ICONS) readonly docIcons: TuiDocIcons,
         @Inject(TUI_COMMON_ICONS) readonly icons: TuiCommonIcons,
+        @Inject(ViewportScroller) readonly viewportScroller: ViewportScroller,
+        @Inject(DOCUMENT) private readonly doc: Document,
     ) {
-        // Angular can't navigate no anchor links
-        // https://stackoverflow.com/questions/36101756/angular2-routing-with-hashtag-to-page-anchor
         title$.subscribe(title => {
             cdr.markForCheck();
             titleService.setTitle(title);
             this.openActivePageGroup();
-            this.handleAnchorLink(this.activatedRoute.snapshot.fragment || '');
         });
+
+        combineLatest([
+            this.router.events.pipe(
+                filter((event): event is Scroll => event instanceof Scroll),
+            ),
+            this.title$.pipe(switchMap(() => this.readyToScroll$.pipe(filter(Boolean)))),
+        ])
+            .pipe(
+                take(1),
+                map(([event]) => event.anchor),
+                filter(Boolean),
+                takeUntil(this.destroy$),
+            )
+            .subscribe(anchor => this.navigateToAnchorLink(anchor as string));
```

Comment:
Why do we need the typecast?

Reply:
<img width="607" alt="image" src="https://github.com/taiga-family/taiga-ui/assets/12021443/ee927136-ceb4-4d45-887c-96db5bf042be">